integration-tests/tests/src/tests/queries.ts

```diff
@@ -15,129 +15,1037 @@
+
+    // TODO: invalid query tests for primitive lists.
   });
 
-  describe("All objects", () => {
-    it("properties and primitive types", () => {
-      expect(persons.length).equal(3);
-      expect(persons[0].name).equal("Alice");
-      expect(persons[0].age).equal(15);
-    });
-
-    it("array of primitive types", () => {
-      expect(contacts.length).equal(3);
-      expect(contacts[0].phones.length).equal(1);
-      expect(contacts[1].phones.length).equal(2);
-      expect(contacts[2].phones.length).equal(0);
-    });
-
-    // https://github.com/realm/realm-js/issues/4844
-    it("emoiji and contains", () => {
-      const text = "unicorn 🦄 today";
-      expect(primitives.length).equal(2);
-      const unicorn1 = primitives.filtered("s CONTAINS 'unicorn 🦄 today'");
-      const unicorn2 = primitives.filtered("s CONTAINS[c] 'unicorn 🦄 today'");
-      const unicorn3 = primitives.filtered("s CONTAINS $0", text);
-      const unicorn4 = primitives.filtered("s CONTAINS[c] $0", text);
-      expect(unicorn1.length).equal(0);
-      expect(unicorn2.length).equal(1);
-      expect(unicorn3.length).equal(0);
-      expect(unicorn4.length).equal(1);
+  describe("Realm Query Language", () => {
+    interface IPrimitive {
+      s: string;
+      b: boolean;
+      i: number;
+      f: number;
+      d: number;
+      t: Date;
+    }
+
+    const PrimitiveSchema: Realm.ObjectSchema = {
+      name: "Primitive",
+      properties: {
+        s: "string",
+        b: "bool",
+        i: "int",
+        f: "float",
+        d: "double",
+        t: "date",
+      },
+    };
+
+    let persons: Realm.Results<IPerson>;
+    let contacts: Realm.Results<IContact>;
+    let primitives: Realm.Results<IPrimitive>;
+
+    openRealmBeforeEach({ schema: [PersonSchema, ContactSchema, PrimitiveSchema] });
+
+    beforeEach(function (this: RealmContext) {
+      this.realm.write(() => {
+        const alice = this.realm.create<IPerson>(PersonSchema.name, { name: "Alice", age: 15 });
+        const bob = this.realm.create<IPerson>(PersonSchema.name, { name: "Bob", age: 14, friends: [alice] });
+        this.realm.create<IPerson>(PersonSchema.name, { name: "Charlie", age: 17, friends: [bob, alice] });
+
+        this.realm.create<IContact>(ContactSchema.name, { name: "Alice", phones: ["555-1234-567"] });
+        this.realm.create<IContact>(ContactSchema.name, { name: "Bob", phones: ["555-1122-333", "555-1234-567"] });
+        this.realm.create<IContact>(ContactSchema.name, { name: "Charlie" });
+
+        this.realm.create<IPrimitive>(PrimitiveSchema.name, {
+          s: "foo",
+          b: true,
+          i: 2,
+          f: 3.14,
+          d: 2.72,
+          t: new Date("2001-05-11T12:45:05"),
+        });
+        this.realm.create<IPrimitive>(PrimitiveSchema.name, {
+          s: "Here is a Unicorn 🦄 today",
+          b: false,
+          i: 44,
+          f: 1.41,
+          d: 4.67,
+          t: new Date("2004-02-26T10:15:02"),
+        });
```

Comment:
Not a part of your work, but this looks a bit verbose and would benefit from something like: https://github.com/realm/realm-js/issues/2009.

Reply:
I am not sure how much less verbose can be made beyond maybe putting them into arrays and doing a forEach / arrayloop, which would hurt readability a bit